server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -747,6 +756,13 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         isRemoteStoreEnabled = settings.getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false);
         remoteStoreRepository = settings.get(IndexMetadata.SETTING_REMOTE_STORE_REPOSITORY);
         isRemoteTranslogStoreEnabled = settings.getAsBoolean(IndexMetadata.SETTING_REMOTE_TRANSLOG_STORE_ENABLED, false);
+        isRemoteSnapshot = IndexModule.Type.REMOTE_SNAPSHOT.match(this);
+        if (isRemoteSnapshot) {
+            Integer minVersion = settings.getAsInt(SEARCHABLE_SNAPSHOT_MINIMUM_VERSION.getKey(), Version.V_EMPTY_ID);
+            if (minVersion > Version.V_EMPTY_ID) {
+                extendedCompatibilitySnapshotVersion = LegacyESVersion.fromId(minVersion);
```

Comment:
Question here please: so for now `extendedCompatibilitySnapshotVersion` is treated as legacy Elasticsearch version, but could it be eventually an OpenSearch version (let say in 3.x it could be set to 1.x)?

Reply:
That's correct! And I'd very much like to improve this code to support that. However, doing so would require heavy refactoring in the `Version` and `LegacyESVersion` classes to support better comparisons between the two. So for now, I've limited this implementation to legacy versions.